playbooks/roles/edxapp/tasks/service_variant_config.yml

```diff
@@ -173,7 +173,7 @@
       owner: "{{ edxapp_user }}"
       group: "{{ common_web_group }}"
       mode:  "0755"
-  when: celery_worker is not defined and not devstack
+  when: celery_worker is not defined and not devstack and EDXAPP_RG_REMOVE_STATICFILES_DIR_DURING_UPDATE
```

Comment:
You change the default behavior of code for all installations. Could you please link a ticket describing why do you do it and how did you test it and ensure that it's not breaking something?

Reply:
this task removes directory which served as `/static/` by nginx - https://github.com/raccoongang/configuration/blob/c6854b5872d9b4a4c272fc3dcf5c1935cd5f14aa/playbooks/roles/nginx/templates/edx/app/nginx/sites-available/static-files.j2#L9

next task (edxapp-update-assets) creates this directory on finish stage and execution timings usually varies from 10 to 20 minutes. within this period LMS and CMS frontend unusable.

Reply:
this task also removes previous versions of "tagged assets" (files with version inserted to filename) which can be referred by browser cached resources

Reply:
Yeah I understand what this task is doing.. I've also read comment above it and it says:
# There are problems with django collectstatic copying files.  It doesn't retain
# last modified timestamps, but relies on those same timestamps to know if a new file
# should be recopied.  While collectstatic --clear exists, it only clears some of the
# files in edxapp_staticfile_dir, it leaves postprocessed or otherwise hashed files.

From my understanding site being down for 10-20 minutes is bad however it seems that If you disable this task we may end up in situation when some part of static files are not rebuilt by django and you'll get a partially broken static and you won't even know about it until someone will complain on it.
This is exactly the reason why this task was created in first place - it's goal is to ensure that 100% of static files are built and no old files remain...

If your goal it to fix 20 minutes downtime issue I think we should take a different approach for now:
1) Build static files in separate directory (/tmp/static for example)
2) When it's ready just delete this folder and put everything from /tmp/static in it's place.
3) Restart services

In such way we're guaranteed not to mess up with some static not rebuild and in the same time reduce overall downtime during upgrade

Reply:
please review latest commit which adds directory with previous version